azure/scope/machinepool.go

```diff
@@ -116,6 +127,45 @@ func NewMachinePoolScope(params MachinePoolScopeParams) (*MachinePoolScope, erro
 	}, nil
 }
 
+// InitMachinePoolCache sets cached information about the machine pool to be used in the scope.
+func (m *MachinePoolScope) InitMachinePoolCache(ctx context.Context) error {
```

Comment:
Changes to this file are adapted from https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3111/files#diff-55bcff341d47dd38d3d18ed5e68d2e07f58aae746f32a58878b445c598b87157R131 to avoid conflicts